HealthcareSystem/PatientWebApp/wwwroot/js/patient_examinations.js

```diff
@@ -654,6 +653,7 @@ function third_step_next() {
                     if (multipleTimes == false) {
                         let appointment = $('<option value="' + i + '">' + appointments[j].startTime + '</option>');
                         appointmentStartTimes.push(appointments[j].startTime);
+                        newAppointments.push(appointment[i]);
```

Comment:
The appointment you add here shouldn't be appointment[i], it should be appointment[j] - the same one you added to the combo box.

Reply:
Fixed in 3596344bef37d835456649237e25d4994465dd14.